### ISSUE TITLE
snapshot statistics over time

### ISSUE THREAD

**Issue by mcmanuss8** [peeringdb/peeringdb#108]
Not sure if this is already available somewhere in the backend, but it might be nice to have monthly snapshots of the database and/or its statistics so one could do month over month trending. 
For example for these stats:

Global System Statistics
8248 Peering Networks
568 Public Exchange Points
18717 Unique Public Exchange Presences
2304 Private Facilities
17030 Private Facility Presences

It'd be nice to show a sparkline next to each which shows how its grown over the past X months or years. In other words - show that peeringdb is growing and by how much. 
Even without a cool visualization, it would be useful for scraping for presentations about peeringdb and the like.

**Comment by arnoldnipper** [comment]
@vegu / @matt could you estimate the work

**Comment by vegu** [comment]
More information is needed before an estimate can be made

Are we to estimate the monthly snapshot and stats aggregation of certain datapoints in the db and also the visualization and public access point of the aggregated data?

- Which datapoints are to be aggregated
- In what manner are the stats to be provided to the public
- If actual visualization of data is also a requirement then design guidelines need to be provided

**Comment by mcmanuss8** [comment]
@vegu  I'd break this ticket into at least two parts:

1. Snapshotting a) the entire db or b) counts within each table. 
2. Visualizing the snapshots.

I think 1a gives the most flexibility but uses more disk space over time. Depends on how resource constrained we are, but the peeringdb cache I have is way less than 1GB so I can't see it being that big to snapshot, say, monthly. This could have ancillary benefits too if we needed to go back and see what a network's value was a year ago for some other report/question.  

2 can be done once 1 is done, and I think it'd be good to do 1 sooner rather than later so we don't lose data. 
For how to do 2, a quick prototype is here: http://jsfiddle.net/bjmwmtzf/1/

**Comment by vegu** [comment]
Objects are already versioned with each change and previous versions could be queried quite easily to aggregate stats (internally, historic data is not available to the public). 

So that still leaves the question which stats should be aggregated, and how to provide them to the public and / or visualize.

_As a side note, since it was brought up, as far as i am aware we are also already doing entire backups of the database regularly, whether or not monthly snapshots are kept i am unsure, @job may know more._

**Comment by mcmanuss8** [comment]
Cool. I wouldn't expect historical data to be public, per se. But it would be useful for the board and committees and others to tout how much growth peeringdb is seeing. 
 
> So that still leaves the question which stats should be aggregated, and how to provide them to the public and / or visualize.

I was thinking all of the stats on the bottom of the page (Global System Statistics). Monthly snapshots of those values would make it non-taxing to get the data to render.

**Comment by arnoldnipper** [comment]
@mcmanuss8 while I see that these figures are interesting you easily are able to get this figures via API. Imho nice to have but definitely on the top of things I want to see implemented

**Comment by mcmanuss8** [comment]
@arnoldnipper  agreed, it's not urgent. Would be nice to have

**Comment by eloos** [comment]
I don't see the return on investment for the community on this one...

**Comment by job** [comment]
up and to the right graphs are somewhat boring, people interested in this data can easily script something themselves :)

**Comment by arnoldnipper** [comment]
As @job said: people interested in data, easily are able to get these data via the API

**Comment by kay23** [comment]
I agree that this is somewhat low priority and it would just show boring up and to the right graphs

**Comment by thatchrisp** [comment]
I agree that this isn't a priority, but it would be nice to have the historical data available somewhere.

**Comment by karumugham** [comment]
No for now. Agree it's nice to have more visualizations, but not pressing. Historical data is already on web.archive.org for anyone who really wants to scrape it and make graphs.
